pulp_docker/app/downloaders.py

```diff
@@ -107,24 +119,31 @@ def __init__(self, *args, **kwargs):
             new_query = parse.urlencode({**parse.parse_qs(parsed_url.query), **auth_query_dict})
             updated_parsed = parsed_url._replace(query=new_query)
             token_url = parse.urlunparse(updated_parsed)
-
-            async with self.session.get(token_url, raise_for_status=True) as token_response:
+            headers = {}
+            if self.remote.username:
+                # for private repos
+                basic = aiohttp.BasicAuth(self.remote.username, self.remote.password).encode()
+                headers['Authorization'] = basic
+            async with self.session.get(token_url, headers=headers, raise_for_status=True) as \
+                    token_response:
                 token_data = await token_response.text()
 
-            self.token['token'] = json.loads(token_data)['token']
+            self.registry_auth['bearer'] = json.loads(token_data)['token']
 
     @staticmethod
-    def auth_header(token):
+    def auth_header(token, basic_auth):
         """
-        Create an auth header that optionally includes a bearer token.
+        Create an auth header that optionally includes a bearer token or basic auth.
 
         Args:
-            token (str): Bearer token to use in header
+            auth (str): Bearer token or Basic auth to use in header
```

Comment:
this should be token and basic_auth

Reply:
actually no, it should one or another. Registry can have whether no auth, token auth or basic auth.
tldr, in the 401 auth header you will see only basic ir bearer scheme.